modules/service/src/main/scala/lucuma/odb/service/GeneratorParamsService.scala

```diff
@@ -118,13 +121,19 @@ object GeneratorParamsService {
 
   extension (mode: InstrumentMode)
     def asImaging(λ: Wavelength): InstrumentMode =
-      mode match {
+      mode match
+        case Flamingos2Spectroscopy(_, f, _)         =>
+          InstrumentMode.Flamingos2Imaging(lucuma.odb.sequence.flamingos2.longslit.Acquisition.toAcquisitionFilter(f))
+        case GmosNorthImaging(_, _)                  =>
+          mode
         case GmosNorthSpectroscopy(_, _, _, _, _, _) =>
           InstrumentMode.GmosNorthImaging(Acquisition.filter(GmosNorthFilter.acquisition, λ, _.wavelength), none)
+        case GmosSouthImaging(_, _)                  =>
```

Comment:
Do we need to include F2 imaging?

Reply:
Yes, we should. We'll need it when we actually do F2 imaging.